### ISSUE TITLE
Edge browser: Buttons in High contrast mode do not render as expected

### ISSUE THREAD

**Issue by moizgh** [OfficeDev/office-ui-fabric-react#8359]
![image](https://user-images.githubusercontent.com/25937675/54548505-76d6b400-497e-11e9-8898-2fed348a1b39.png)

This is what the states should look like for the two types of buttons, but once key focus hits the standard button, hover is broken. The correct visual is above the third column

**Comment by moizgh** [comment]
The Standard button in the third column of screenshot seems to be missing -ms-high-contrast-adjust: none styling which is making the text disappear.

**Comment by micahgodbolt** [comment]
@moizgh if you know the fix, you are welcome to create a PR for the component. That's the quickest way to get this resolved.

**Comment by khmakoto** [comment]
Now this is weird. This seems to repro on the official site, but I cannot get it to repro on my local build nor on codepen, as demonstrated [here](https://codepen.io/khmakoto/pen/EMpgbz).

@Jahnp @dzearing @micahgodbolt do any of you have an idea of what might be happening here? Seems like it's an issue with the site.

For reference:

__In the official site:__
![image](https://user-images.githubusercontent.com/7798177/54569230-48190780-4997-11e9-81d0-e4bfc76e63c5.png)

__In the local build:__
![image](https://user-images.githubusercontent.com/7798177/54569245-55ce8d00-4997-11e9-92bd-79970ca8bedd.png)

__In a codepen:__
![image](https://user-images.githubusercontent.com/7798177/54569276-75fe4c00-4997-11e9-8e56-0fae9e47db35.png)

**Comment by dzearing** [comment]
you may need to inspect which rules are causing the problem. They could be coming form the header.

**Comment by khmakoto** [comment]
After taking a deep dive it seems that the error is being caused by the inclusion of `https://mwf-service.akamaized.net/mwf/css/bundle/1.55.0/west-european/default/mwf-main.min.css` in the main site, which is included in a `link` tag at the beginning of the `body` tag. I assume it applies some styling that screws with the `HighContrast` styling of our `Buttons`.

@dzearing what approach should we take here?

**Comment by micahgodbolt** [comment]
we certainly don't want to include anything in our button bundle to override this...maybe we could add it to the page css via this 
![image](https://user-images.githubusercontent.com/1434956/54571311-697df180-499f-11e9-8761-cc12e9e120c9.png)

**Comment by khmakoto** [comment]
The thing is that it's breaking more than just the examples. The `Show code` buttons are having a similar behavior, so they would also be broken even if we fix this for the examples.

**Comment by micahgodbolt** [comment]
ah joy....we might have to go fix those show code buttons individually.

**Comment by khmakoto** [comment]
The other thing is that this is going to be a pain to fix because this is an issue with the live site only, so we won't have any indication of whether any solution we try is working or not until the live site is updated daily.

**Comment by KevinTCoughlin** [comment]
Can we recreate the greedy selector locally? Its usually available via the "Computed" tab in the browser. Also, IIRC the team is receptive to relaxing whatever rule is conflicting.

**Comment by khmakoto** [comment]
@KevinTCoughlin I gave some thought to it and I'll try to recreate the selector. This fix will probably take some time, however, so I'm thinking of doing this next week when I'm on secondary.

**Comment by moizgh** [comment]
@khmakoto 
Any updates on this?

**Comment by khmakoto** [comment]
@moizgh I was able to recreate the selector but not override it. There was some discussion with @Jahnp about this which mostly concluded in communicating with the team managing the headers iirc to get to a solution but it certainly seemed that it would take some time, especially with all the changes that were ongoing on the website.

@Jahnp do you know what the next steps are for this problem?

**Comment by AndrewCraswell** [comment]
I just came across this issue as the same was flagged on our site during accessibility review. Two things here:

1. As noted above, the documentation site has MWF styles overriding Fabric styles which made it hard to identify the root issue.
2. I created a [Codepen](https://codepen.io/andrew-craswell/pen/JgLRLz?&editable=true) with the buttons without the MWF code and can still see some issues. On Windows 10, Edge browser in Black and White high-contrast mode the Visual Focus Indicator is not visible for the `PrimaryButton` component.

For reference, the below image shows what the Primary button looks like in high-contrast mode when focused. It doesn't appear visually any different from when it's not focused.

![Example of Primary button selected](https://i.imgur.com/RQggGLR.png "Example of Button")

**Comment by AndrewCraswell** [comment]
No, I'm saying that in the codepen even without MWF, the VFI can't be seen.

**Comment by micahgodbolt** [comment]
@AndrewCraswell can you provide a codepen that repros the issue. From @khmakoto's comment, it works fine in codepen but is broken on our site (which is caused by MWF styles).

**Comment by AndrewCraswell** [comment]
I might have read the original issue incorrectly. What I was saying is that the Visual Focus Indicator does not display in Black/White high-contrast mode, which is what my [Codepen](https://codepen.io/andrew-craswell/pen/JgLRLz?&editable=true) above showed.

**Comment by micahgodbolt** [comment]
Ok, I noted this specific problem about focus in high contrast here #10299.

I'm going to close this issue down because the original issue is caused by 3rd party css on the page, and this discussion is getting a bit fragmented.
